s3/pkg/helper/functional.go

```diff
@@ -0,0 +1,19 @@
+package helper
+
+func Filter(xs []string, f func(string) bool) []string {
```

Comment:
for similar code [online](https://stackoverflow.com/questions/37562873/most-idiomatic-way-to-select-elements-from-an-array-in-golang), found this Note below, to optimize
Note:

> If it is expected that many elements will be selected, it might be profitable to allocate a "big" ret slice beforehand, and use simple assignment instead of the append(). And before returning, slice the ret to have a length equal to the number of selected elements.